src/ray/raylet/main.cc

```diff
@@ -54,14 +55,24 @@ int main(int argc, char *argv[]) {
   object_manager_config.store_socket_name = store_socket_name;
   object_manager_config.pull_timeout_ms =
       RayConfig::instance().object_manager_pull_timeout_ms();
-  object_manager_config.max_sends = RayConfig::instance().object_manager_max_sends();
-  object_manager_config.max_receives =
-      RayConfig::instance().object_manager_max_receives();
   object_manager_config.push_timeout_ms =
       RayConfig::instance().object_manager_push_timeout_ms();
+
+  // This may be 0 when core detection fails.
+  int num_cores = std::thread::hardware_concurrency();
+  object_manager_config.max_sends = std::max(1, num_cores / 4);
```

Comment:
Let's leave this configurable and initialize with this variable in the RayConfig. Also we should use num_cpus as in https://github.com/ray-project/ray/commit/e1024d84e92b2596958037dc411795b6639ef08c and not the hardware_concurrency (first because a user may want to use only a fraction of a machine for ray and second because std::thread::hardware_concurrency seems to be less portable, as you point out it could even be 0).